cairo/ethereum/cancun/vm/interpreter.cairo

```diff
@@ -8,27 +8,211 @@ from starkware.cairo.common.math_cmp import is_nn
 from ethereum.cancun.vm import Evm, EvmStruct, Message, Environment, EvmImpl, EnvImpl
-from ethereum.cancun.vm.exceptions import EthereumException, StackDepthLimitError, Revert
+from ethereum.cancun.vm.exceptions import (
+    EthereumException,
+    InvalidContractPrefix,
+    OutOfGasError,
+    StackDepthLimitError,
+    Revert,
+)
+from ethereum.cancun.vm.instructions import op_implementation
+from ethereum.cancun.vm.memory import Memory, MemoryStruct, Bytes1DictAccess
+from ethereum.cancun.vm.runtime import get_valid_jump_destinations
+from ethereum.cancun.vm.stack import Stack, StackStruct, StackDictAccess
+from ethereum.utils.numeric import U256, U256Struct, U256__eq__
 from ethereum.cancun.state import (
     State,
     begin_transaction,
+    destroy_storage,
+    mark_account_created,
+    increment_nonce,
     commit_transaction,
     rollback_transaction,
-    touch_account,
     move_ether,
+    touch_account,
+    set_code,
 )
-from ethereum.cancun.vm.instructions import op_implementation
-from ethereum.cancun.vm.memory import Memory, MemoryStruct, Bytes1DictAccess
-from ethereum.cancun.vm.runtime import get_valid_jump_destinations
-from ethereum.cancun.vm.stack import Stack, StackStruct, StackDictAccess
-from ethereum.utils.numeric import U256, U256Struct, U256__eq__
 
 from src.utils.dict import dict_new_empty
 
 const STACK_DEPTH_LIMIT = 1024;
+const MAX_CODE_SIZE = 0x6000;
+
+func process_create_message{
+    range_check_ptr,
+    bitwise_ptr: BitwiseBuiltin*,
+    keccak_ptr: KeccakBuiltin*,
+    poseidon_ptr: PoseidonBuiltin*,
+}(message: Message, env: Environment) -> (Evm, EthereumException*) {
+    alloc_locals;
+
+    let state = env.value.state;
+    let transient_storage = env.value.transient_storage;
+
+    // take snapshot of state before processing the message
+    begin_transaction{state=state, transient_storage=transient_storage}();
+
+    // If the address where the account is being created has storage, it is
+    // destroyed. This can only happen in the following highly unlikely
+    // circumstances:
+    // * The address created by a `CREATE` call collides with a subsequent
+    //   `CREATE` or `CREATE2` call.
+    // * The first `CREATE` happened before Spurious Dragon and left empty
+    //   code.
+    destroy_storage{state=state}(message.value.current_target);
+
+    // In the previously mentioned edge case the preexisting storage is ignored
+    // for gas refund purposes. In order to do this we must track created
+    // accounts.
+    mark_account_created{state=state}(message.value.current_target);
+
+    increment_nonce{state=state}(message.value.current_target);
+    EnvImpl.set_state{env=env}(state);
+    EnvImpl.set_transient_storage{env=env}(transient_storage);
+    let (evm, err) = process_message(message, env);
+    if (cast(err, felt) != 0) {
+        return (evm, err);
+    }
+
+    // Success case
+    if (cast(evm.value.error, felt) == 0) {
```

Comment:
Should error case be handled first with early return?
It will be easier to read and avoid a big if